src/CoreWCF.Http/src/CoreWCF/Security/SecurityUtils.cs

```diff
@@ -16,5 +17,16 @@ public static void ValidateAnonymityConstraint(WindowsIdentity identity, bool al
                     new SecurityTokenValidationException(SR.AnonymousLogonsAreNotAllowed));
             }
         }
+
+        public static bool ShouldUseAuthentication(Binding binding)
+        {
+            if (binding is WSHttpBinding wsHttpBinding
+                && wsHttpBinding.Security.Mode != SecurityMode.None
```

Comment:
This isn't just needed for WSHttpBinding. Also you might have a CustomBinding which is equivalent. You need to call CreateBindingElements and on the resulting collection, use `Find<HttpTransportBindingElement>()` to get the transport binding element. Then you can check AuthenticationScheme to see if authentication is required.